x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/script.tsx

```diff
@@ -0,0 +1,186 @@
+      defaultMessage: 'Source',
+    }),
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.scriptForm.sourceFieldHelpText',
+      {
+        defaultMessage: 'Script to be executed.',
+      }
+    ),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate('xpack.ingestPipelines.pipelineEditor.scriptForm.sourceRequiredError', {
+            defaultMessage: 'A value is required.',
+          })
+        ),
+      },
+    ],
+  },
+
+  /* Optional fields config */
+  lang: {
+    type: FIELD_TYPES.TEXT,
+    deserializer: String,
+    serializer: from.undefinedIfValue('painless'),
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.scriptForm.langFieldLabel', {
+      defaultMessage: 'Language (optional)',
+    }),
+    helpText: (
+      <FormattedMessage
+        id="xpack.ingestPipelines.pipelineEditor.scriptForm.langFieldHelpText"
+        defaultMessage="Script language. Defaults to {lang}."
+        values={{
+          lang: <EuiCode>{'painless'}</EuiCode>,
+        }}
+      />
+    ),
+  },
+
+  params: {
+    type: FIELD_TYPES.TEXT,
+    deserializer: to.jsonString,
+    serializer: from.optionalJson,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.scriptForm.paramsFieldLabel', {
+      defaultMessage: 'Parameters',
+    }),
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.scriptForm.paramsFieldHelpText',
+      {
+        defaultMessage: 'Script parameters.',
+      }
+    ),
+    validations: [
+      {
+        validator: (value) => {
+          if (value.value) {
+            return isJsonField(
+              i18n.translate(
+                'xpack.ingestPipelines.pipelineEditor.scriptForm.processorInvalidJsonError',
+                {
+                  defaultMessage: 'Invalid JSON',
+                }
+              )
+            )(value);
+          }
+        },
+      },
+    ],
+  },
+};
+
+export const Script: FormFieldsComponent = ({ initialFieldValues }) => {
+  const [showId, setShowId] = useState(() => !!initialFieldValues?.id);
+  return (
+    <>
+      <EuiFormRow>
+        <EuiSwitch
+          label={i18n.translate(
+            'xpack.ingestPipelines.pipelineEditor.scriptForm.useScriptIdToggleLabel',
+            { defaultMessage: 'Use stored script' }
+          )}
+          checked={showId}
+          onChange={() => setShowId((v) => !v)}
+        />
+      </EuiFormRow>
+
+      {showId ? (
+        <UseField key="fields.id" path="fields.id" component={Field} config={fieldsConfig.id} />
+      ) : (
+        <>
+          <UseField component={Field} config={fieldsConfig.lang} path="fields.lang" />
+
+          <UseField
+            key="fields.source"
+            path="fields.source"
+            component={TextEditor}
+            componentProps={{
+              editorProps: {
+                height: 100,
```

Comment:
I'm glad you called this out! These editor sizes can definitely benefit from some standardisation. What I tried to do was consider how much text I was expecting in each case and give the editor a height that more or less communicates that expectation. It would be good to reflect this in the code too!

Reply:
But if it is an issue we should fix it in Painless lab too! 😅